Code/network/include/Connection.h

```diff
@@ -79,4 +93,7 @@ class Connection
     uint64_t m_timeSinceLastEvent;
     Endpoint m_remoteEndpoint;
     DHChachaFilter m_filter;
+    bool m_isServer;

```

Comment:
Kinda sucks to store all this stuff for alive connections, might be best to have two other classes, ClientNegociation and ServerNegociation that would do this and then create a Client

Reply:
Sure, but if we want our disconnection packets authenticated we'd still have to store a code per connection...